l3kernel/l3tl.dtx

```diff
@@ -2792,13 +2851,12 @@
   }
 \cs_new:Npn \@@_act_normal:NwnNNN #1 #2 \q_@@_act_stop #3#4
   {
-    \if_meaning:w \q_@@_act_mark #1
-      \exp_after:wN \@@_act_end:wn
-    \fi:
+    \@@_use_none_delimit_by_act_mark:w #1 \@@_act_end:wn \q_@@_act_mark
     #4 {#3} #1
     \@@_act_loop:w #2 \q_@@_act_stop
     {#3} #4
   }
+\cs_new:Npn \@@_use_none_delimit_by_act_mark:w #1 \q_@@_act_mark { }
```

Comment:
I think it should be "`by_q_act_mark`".

Reply:
This one is true, but I'm not sure whether it then should be "`by_q__tl_act_mark`" (which on the other hand looks weird for the double underscore), instead.

Reply:
whichever way you turn it, is is a bit odd. Perhaps be imprecise and call it `\@@_use_none_delimit_by_quark:w`.

Reply:
@FrankMittelbach the version suggested by @blefloch is equivalent to the public ones in `l3basics` (except for being private to `l3tl` hinting at the argument being delimited by a private quark). I had these thoughts when I finally settled for leaving out the `q` instead to make it less ambiguous and circumventing the issue with the double underscore in the macro name. I have no strong opinion on this though, so more input by the team (or a definitive decision by someone) would be welcome.

Reply:
The question already arose when Phelype made the switch to mostly internal quarks.  See in `l3basics`

    \cs_new:Npn \@@_use_i_delimit_by_q_recursion_stop:nw
      #1 #2 \q_@@_recursion_stop {#1}

Reply:
Oh, I wasn't aware of it, thank you for pointing it out!

Reply:
I have no strong feelings either way (my suggestion was just a side remark), What I don't like though would be double underscores in the middle of names, if we w have already established a convention elsewhere let's stick to that convention as suggested by Bruno and Phelype.